### PR TITLE
Basic Auth Implementation

### PR DESCRIPTION
**Pull request by touhidurrr**

### CODE REVIEW

**Comment by touhidurrr** [comment]
@GGGuenni, can you review this?

**Comment by touhidurrr** [comment]
@oynqr also

File: src/plugins/MongoDB.ts [hunk 1]
```diff
@@ -49,6 +56,7 @@ export default fp(async function MongoDB(server) {
   await mongoClient.connect();
 
   const db = {
+    Auth: await mongoClient.db('unciv').collection<Auth>('ErrorLogs'),
```
Comment (GGGuenni) [inline src/plugins/MongoDB.ts:59]: The `ErrorLogs` looks like a copy mistake?

**Review by GGGuenni — COMMENTED** [review]
Looks good in general(?) I've never used mongoDB, redis nor typescript or fastify before so I'm probably not the right person to review this. 
If you tested it, it should be fine. Worst case multiplayer doesn't work for a short amount of time which unciv players have a lot of experience with 😄

File: src/plugins/MongoDB.ts (hunk 1 above)
Comment (touhidurrr) [reply src/plugins/MongoDB.ts]: yeah, fixed and almost everything rewritten

**Comment by touhidurrr** [comment]
@GGGuenni @oynqr, I have done a lot of tests on it and found no problems.
Will merge it when @GGGuenni fixes unintentional Gzip on the `PUT /auth` request body.
Do tell me if you notice any errors in the meantime!!
Also @oynqr I am using `argon2id` for this pr.

**Comment by touhidurrr** [comment]
I think with https://github.com/touhidurrr/UncivServer.xyz/pull/5/commits/dd4a2443d3f104cda0acff056141e8d55e70ecbc, we can merge this.

**Pull request merged**